src/preset.ts

```diff
@@ -143,8 +144,14 @@ export function evaluateCondition(condition: Condition): boolean {
 }
 
 function evaluateInheritedPresetConditions(preset: Preset, allPresets: Preset[], references: Set<string>): boolean | undefined {
-    const evaluateParent = (parentName: string) => {
+    const evaluateParent = (parentName: string, isChildUserPreset?: boolean) => {
```

Comment:
You shouldn't need to add a variable for this.  You have access to `preset.isUserPreset` in this function which is the same thing.